src/java.base/share/classes/java/text/ChoiceFormat.java

```diff
@@ -114,42 +119,62 @@
  * }
  * }
  * </blockquote>
- * <p>
- * Specifying a pattern for ChoiceFormat objects is fairly straightforward.
- * For example:
+ * Would output the following:
+ * <blockquote>
+ * <pre>{@code
+ * There are no files on ADisk
+ * There is one file on ADisk
+ * There are 2 files on ADisk
+ * There are 3 files on ADisk
+ * }</pre>
+ * </blockquote>
+ *
+ * <h2><a id="patterns">Patterns</a></h2>
+ * A {@code ChoiceFormat} pattern has the following syntax:
+ * <blockquote><pre>
+ * <i>Pattern:</i>
+ *         SubPattern *("|" SubPattern)
+ *         <i>Note: Each additional subPattern must have a limit greater than the previous subPattern's limit</i>
+ * <i>SubPattern:</i>
+ *         Limit Relation Format
+ * <i>Limit:</i>
+ *          Number / "&infin;" / "-&infin;"
+ * <i>Number:</i>
+ *         *(Digit) 1*(Decimal / Digit) *(Digit) [Exponent]
+ * <i>Decimal:</i>
+ *         1*(Digit ".") / 1*("." Digit)
+ * <i>Digit:</i>
+ *         0 - 9
+ * <i>Exponent:</i>
+ *         *(Digit) Digit ExponentSymbol Digit *(Digit)
+ * <i>ExponentSymbol:</i>
+ *         "e" / "E"
+ * <i>Relation:</i>
+ *         "#" / "&lt;" / "&le;"
+ * <i>Format:</i>
+ *         Any characters except the <i>Relation</i> symbols
+ * </pre></blockquote>
+ *
+ * <i>Note:The relation &le; is not equivalent to &lt;&equals;</i>
+ *
+ * <p>Below is an example of constructing a ChoiceFormat with a pattern:
```

Comment:
FWIW, in other parts of core libs such as java.lang.Double, I've used a definition list `<dl>` with terms `<dt>` and definitions `<dd>` to represent such grammars.